src/Adapter/Product/Repository/CombinationRepository.php

```diff
@@ -0,0 +1,288 @@
+        ;
+
+        return (int) $qb->execute()->fetch()['total_combinations'];
+    }
+
+    /**
+     * @param int[] $combinationIds
+     * @param LanguageId $langId
+     *
+     * @return array<int, array<int, mixed>>
+     */
+    public function getAttributesInfoByCombinationIds(array $combinationIds, LanguageId $langId): array
+    {
+        $attributeCombinationAssociations = $this->getAttributeCombinationAssociations($combinationIds);
+
+        $attributeIds = array_unique(array_map(function (array $attributeByCombination): int {
+            return (int) $attributeByCombination['id_attribute'];
+        }, $attributeCombinationAssociations));
+
+        $attributesInfoByAttributeId = $this->getAttributesInformation($attributeIds, $langId->getValue());
+
+        $attributesInfoByCombinationId = [];
+        foreach ($attributeCombinationAssociations as $attributeCombinationAssociation) {
+            $combinationId = (int) $attributeCombinationAssociation['id_product_attribute'];
+            $attributeId = (int) $attributeCombinationAssociation['id_attribute'];
+            $attributesInfoByCombinationId[$combinationId][] = $attributesInfoByAttributeId[$attributeId];
+        }
+
+        return $attributesInfoByCombinationId;
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     *
+     * @throws CoreException
+     */
+    public function assertCombinationExists(CombinationId $combinationId): void
+    {
+        $this->assertObjectModelExists(
+            $combinationId->getValue(),
+            'product_attribute',
+            CombinationNotFoundException::class
+        );
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     * @param int[] $attributeIds
+     */
+    public function saveProductAttributeAssociation(CombinationId $combinationId, array $attributeIds): void
+    {
+        $this->assertCombinationExists($combinationId);
+        $this->attributeRepository->assertAllAttributesExist($attributeIds);
+
+        $attributesList = [];
+        foreach ($attributeIds as $attributeId) {
+            $attributesList[] = [
+                'id_product_attribute' => $combinationId->getValue(),
+                'id_attribute' => $attributeId,
+            ];
+        }
+
+        try {
+            if (!Db::getInstance()->insert('product_attribute_combination', $attributesList)) {
+                throw new CannotAddCombinationException('Failed saving product-combination associations');
+            }
+        } catch (PrestaShopException $e) {
+            throw new CoreException('Error occurred when saving product-combination associations', 0, $e);
+        }
+    }
+
+    /**
+     * @param int[] $combinationIds
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getAttributeCombinationAssociations(array $combinationIds): array
+    {
+        $qb = $this->connection->createQueryBuilder();
```

Comment:
I just realized this class uses a mix of ObjectModel / Db and Doctrine 🤔

Reply:
as mentioned in slack, its only due simplicity (so I don't need to write my own sql query when there is one written and working already). Legacy already provided method to insert multiple product attributes, so I just reused it